src/execution/StepExecutionManager.ts

```diff
@@ -62,7 +70,72 @@ export class StepExecutionManager {
     let process = statusManager.findOrCreateProcess(step, currentProcessType)
 
     if (process.status !== 'DONE') {
+      const multisigProcess = step.execution.process.find(
+        (p) => !!p.multisigTxHash
+      )
+
       try {
+        if (isMultisigSigner && multisigProcess) {
+          if (!multisigProcess) {
+            throw new Error('MultiSig process is undefined')
+          }
+          if (!config.multisigConfig?.getMultisigTransactionDetails) {
+            throw new Error(
+              '"getMultiSigTransactionDetails()" is missing in MultiSig config.'
+            )
+          }
+
+          const multisigTxHash = multisigProcess.multisigTxHash
+
+          if (!multisigTxHash) {
+            // need to check what happens in failed tx
+            throw new Error('Multisig internal transaction hash is undefined.')
+          }
+
+          const response: MultiSigTxDetails =
+            await config.multisigConfig?.getMultisigTransactionDetails(
+              multisigTxHash
+            )
+
+          if (response.status === 'PENDING') {
+            return step.execution!
+          }
+
+          if (response.status === 'SUCCESS') {
+            process = statusManager.updateProcess(
+              step,
+              process.type,
+              'PENDING',
+              {
+                txHash: response.txHash,
+                multisigTxHash: undefined,
+                txLink:
+                  fromChain.metamask.blockExplorerUrls[0] +
+                  'tx/' +
+                  response.txHash,
+              }
+            )
+          }
+
+          if (response.status === 'FAILED') {
+            throw new TransactionError(
+              LifiErrorCode.TransactionFailed,
+              '',
+              '',
+              ''
```

Comment:
Why is this empty? 🙂

Reply:
umm, I didn't exactly know the error to fill in, I'll write the same message itself. Lmk if it works